File: api_deploy/backend/create_super_admin.py
import asyncio
from sqlalchemy import select
from app.core.database import async_session_maker
from app.core.security import hash_password
from app.models import User


async def create_super_admin():
    async with async_session_maker() as db:
        result = await db.execute(
            select(User).where(User.email == "superadmin@smarttutor.com")
        )
        existing = result.scalar_one_or_none()

        if existing:
            print("Super admin already exists!")
            return

        super_admin = User(
            username="superadmin",
            email="superadmin@smarttutor.com",
            hashed_password=hash_password("SuperAdmin123!"),
            full_name="Super Admin",
            role="super_admin",
            is_active=True,
            is_verified=True,
        )
        db.add(super_admin)
        await db.commit()
        print("Super admin created successfully!")
        print("Email: superadmin@smarttutor.com")
        print("Password: SuperAdmin123!")


if __name__ == "__main__":
    asyncio.run(create_super_admin())

File: backend-node/scripts/createSuperAdmin.ts
/**
 * Super admin upsert — Node port of backend/create_super_admin.py +
 * backend/fix_super_admin.py combined (create if missing, else reset
 * password to the documented credential).
 *
 * Usage: npm run superadmin
 */
import { pool } from "../src/db.js";
import { hashPassword } from "../src/security.js";

const EMAIL = "superadmin@smarttutor.com";
const PASSWORD = "SuperAdmin123!";

async function main(): Promise<void> {
  const existing = await pool.query<{ id: string }>(
    `SELECT id FROM users WHERE email = $1`,
    [EMAIL]
  );
  if (existing.rows[0]) {
    await pool.query(
      `UPDATE users SET hashed_password = $2, is_active = true, is_verified = true WHERE id = $1`,
      [existing.rows[0].id, hashPassword(PASSWORD)]
    );
    console.log("Super admin password reset to the documented credential.");
  } else {
    await pool.query(
      `INSERT INTO users (username, email, hashed_password, full_name, role, is_active, is_verified)
       VALUES ($1, $2, $3, $4, 'super_admin', true, true)`,
      ["superadmin", EMAIL, hashPassword(PASSWORD), "Super Admin"]
    );
    console.log("Super admin created.");
  }
  console.log(`Email: ${EMAIL}`);
  console.log(`Password: ${PASSWORD}`);
  await pool.end();
}

main().catch((err) => {
  console.error(err instanceof Error ? err.message : err);
  process.exit(1);
});